L1Trigger/L1THGCal/src/backend/HGCalHistoSeedingImpl.cc

```diff
@@ -49,16 +58,26 @@ HGCalHistoSeedingImpl::Histogram HGCalHistoSeedingImpl::fillHistoClusters(
     for (int bin_R = 0; bin_R < int(nBinsRHisto_); bin_R++) {
       for (int bin_phi = 0; bin_phi < int(nBinsPhiHisto_); bin_phi++) {
         histoClusters[{{z_side, bin_R, bin_phi}}] = 0;
+	averageSeedPositions_[{{z_side, bin_R, bin_phi}}] = std::make_pair(0,0); 
       }
     }
   }
 
+
+
   for (auto& clu : clustersPtrs) {
     float ROverZ = sqrt(pow(clu->centreProj().x(), 2) + pow(clu->centreProj().y(), 2));
     int bin_R = int((ROverZ - kROverZMin_) * nBinsRHisto_ / (kROverZMax_ - kROverZMin_));
     int bin_phi = int((reco::reduceRange(clu->phi()) + M_PI) * nBinsPhiHisto_ / (2 * M_PI));
 
     histoClusters[{{triggerTools_.zside(clu->detId()), bin_R, bin_phi}}] += clu->mipPt();
+    averageSeedPositions_[{{triggerTools_.zside(clu->detId()), bin_R, bin_phi}}].first += ( clu->centreProj().x() ) * clu->mipPt();  
```

Comment:
To avoid searching multiple times in the map, a small `struct` (e.g. named `Bin`) containing the bin information (`mipPt` sum, `x`, `y`) could be defined, and the `Histogram` redefined as `typedef std::map<std::array<int, 3>, Bin> Histogram;`